include/envoy/server/instance.h

```diff
@@ -145,6 +145,11 @@ class Instance {
   virtual void shutdownAdmin() PURE;
 
   /**
+   * @return Singleton::Manager& the server-wide singleton manager.
+   */
+  virtual Singleton::Manager& singletonManager() PURE;
```

Comment:
Thanks for the clarification. Yeah, either an issue or maybe just some TODO to move the other ones under here later.